### PR TITLE
JAMES-2801 use elastic Scheduler by default for retry and delay

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

**Comment by rouazana** [comment]
Can you explain the rationals behind these modifications?

**Comment by Arsnael** [comment]
Read it...

**Comment by chibenwa** [comment]
test this please

File: backends-common/rabbitmq/src/main/java/org/apache/james/backend/rabbitmq/RabbitMQConnectionFactory.java [hunk 1]
```diff
@@ -58,8 +58,8 @@ Connection create() {
     }
 
     Mono<Connection> connectionMono() {
+        Duration forever = Duration.ofMillis(Long.MAX_VALUE);
```
Comment (trantienduchn) [inline backends-common/rabbitmq/src/main/java/org/apache/james/backend/rabbitmq/RabbitMQConnectionFactory.java]: it should be extracted into a constant, and somewhere else

**Comment by mbaechler** [comment]
> Can you explain the rationals behind these modifications?

I did wrote the rational into the ticket: https://issues.apache.org/jira/browse/JAMES-2801

TL;DR
Retry switch your processing in parallel and it's not what we want most of the time.

File: backends-common/rabbitmq/src/main/java/org/apache/james/backend/rabbitmq/RabbitMQConnectionFactory.java (hunk 1 above)
Comment (mbaechler) [reply backends-common/rabbitmq/src/main/java/org/apache/james/backend/rabbitmq/RabbitMQConnectionFactory.java]: For what? Is forever ever going to change?
Are we going to use this FOREVER for any other purpose? Are we going to reference it every time we call retryBackoff? And what if we fail? Is it a problem?
In my opinion, defining this constant would require a time module or a reactor module somewhere and we would then include it everywhere. And then, we would have to know that we should not use the reactor code directly but use this thin wrapper we have. It looks like something I don't want to do for now for a single constant that will likely never change.

**Comment by mbaechler** [comment]
```
[fa54f12fea2d7390419baa813f47ed6bcfa62f0f] [ERROR] Errors: 
[fa54f12fea2d7390419baa813f47ed6bcfa62f0f] [ERROR]   LinshareBlobExportMechanismProvidingTest.<clinit>:29 » ExceptionInInitializer
```

test this please

**Comment by trantienduchn** [comment]
```
[fa54f12fea2d7390419baa813f47ed6bcfa62f0f] [ERROR] Errors: 
[fa54f12fea2d7390419baa813f47ed6bcfa62f0f] [ERROR]   FilterTest.messageShouldBeAppendedInSpecificMailboxWhenFromRuleMatches:749 » ConditionTimeout
```
test this please

File: backends-common/rabbitmq/src/main/java/org/apache/james/backend/rabbitmq/RabbitMQConnectionFactory.java (hunk 1 above)
Comment (trantienduchn) [reply backends-common/rabbitmq/src/main/java/org/apache/james/backend/rabbitmq/RabbitMQConnectionFactory.java]: I just meant, extract it into a private constant in that class to save a `Duration.ofMillis(Long.MAX_VALUE)` call each time calling connectionMono()

Comment (mbaechler) [reply backends-common/rabbitmq/src/main/java/org/apache/james/backend/rabbitmq/RabbitMQConnectionFactory.java]: Let's keep it simple for now, I'm pretty sure it will never be a performance problem.
